tests/integration/conftest.py

```diff
@@ -84,24 +84,39 @@ def service_instance(integration_test_config):
 @pytest.fixture
 def patch_salt_globals(vmware_conf):
     """
-    Patch __opts__ and __pillar__
+    Patch __opts__, __pillar__ and __salt__
     """
     setattr(datacenter_mod, "__opts__", {})
     setattr(datacenter_mod, "__pillar__", {})
+    setattr(datacenter_mod, "__salt__", vmware_conf)
+    setattr(datacenter_st, "__salt__", vmware_conf)
     setattr(cluster_mod, "__opts__", {})
     setattr(cluster_mod, "__pillar__", {})
+    setattr(cluster_mod, "__salt__", vmware_conf)
     setattr(cluster_ha_mod, "__opts__", {})
     setattr(cluster_ha_mod, "__pillar__", {})
+    setattr(cluster_ha_mod, "__salt__", vmware_conf)
     setattr(cluster_drs_mod, "__opts__", {})
     setattr(cluster_drs_mod, "__pillar__", {})
+    setattr(cluster_drs_mod, "__salt__", vmware_conf)
     setattr(esxi_mod, "__pillar__", vmware_conf)
     setattr(esxi_mod, "__opts__", {})
+    setattr(esxi_mod, "__salt__", vmware_conf)
     setattr(esxi_st, "__pillar__", vmware_conf)
+    setattr(esxi_st, "__salt__", vmware_conf)
```

Comment:
**suggestion** it's not necessarily the responsibility for this function, but this fixture is entirely wrong. Either we should access these via the salt loader (make calls via the salt_call_cli), or  we should use the `configure_loader_modules` fixture, which ensures that faked attributes like this are cleaned up.

Reply:
I'm not sure about that at all or even how to fix it. I just followed the convention I saw here.

Reply:
Oooo... now I see. Yeah, we can remove `__salt__` from here for sure.